.github/workflows/ci-build.yml

```diff
@@ -1,22 +1,112 @@
+
+      # Prep for running the playbook
+      - name: Create hosts file
+        run: echo -e "[icat-test-hosts]\nlocalhost ansible_connection=local" > icat-ansible/hosts
+      - name: Prepare vault pass
+        run: echo -e "icattravispw" > icat-ansible/vault_pass.txt
+      - name: Move vault to directory it'll get detected by Ansible
+        run: mv icat-ansible/vault.yml icat-ansible/group_vars/all
+      - name: Replace default payara user with Actions user
+        run: |
+          sed -i -e "s/^payara_user: \"glassfish\"/payara_user: \"runner\"/" icat-ansible/group_vars/all/vars.yml
+      - name: Amending roles
+        run: |
+          sed -i 's/role: authn-uows-isis/role: authn-anon/' icat-ansible/icat-test-hosts.yml
+
+      # Force hostname to localhost - bug fix for previous ICAT Ansible issues on Actions
+      - name: Change hostname to localhost
+        run: sudo hostname -b localhost
+
+      # Create local instance of ICAT
+      - name: Run ICAT Ansible Playbook
+        run: |
+          ansible-playbook icat-ansible/icat-test-hosts.yml -i icat-ansible/hosts --vault-password-file icat-ansible/vault_pass.txt -vv
+
+      - name: Checkout datagateway-api
+        uses: actions/checkout@v2
+        with:
+          repository: ral-facilities/datagateway-api
+          path: datagateway-api
+
+      # DataGateway API file setup
+      - name: Create log file
+        run: touch logs.log
+      - name: Configure log file location
+        run: echo "`jq -r --arg REPO_DIR "$GITHUB_WORKSPACE/logs.log" \
+          '.log_location=$REPO_DIR' datagateway-api/datagateway_api/config.json.example`" > datagateway-api/datagateway_api/config.json.example
+      - name: Configure API backend
+        run: echo "`jq -r --arg BACKEND "python_icat" \
+          '.backend=$BACKEND' datagateway-api/datagateway_api/config.json.example`" > datagateway-api/datagateway_api/config.json.example
+      - name: Set debug mode to true
+        run: |
+          sed -i -e "s/\"debug_mode\": false,/\"debug_mode\": true,/" datagateway-api/datagateway_api/config.json.example
+      - name: Set log level to DEBUG
+        run: |
+          sed -i -e "s/\"log_level\": \"WARN\",/\"log_level\": \"DEBUG\",/" datagateway-api/datagateway_api/config.json.example
+      - name: Create config.json
+        run: cp datagateway-api/datagateway_api/config.json.example datagateway-api/datagateway_api/config.json
+      
+      # DataGateway API dependencies
+      - name: Install Poetry
+        run: pip install poetry
+      - name: Install dependencies
+        run: cd datagateway-api; poetry install; cd ..
+      - name: Add dummy data to icatdb
+        run: |
+          cd datagateway-api; poetry run python -m util.icat_db_generator; cd ..
+
+      - name: Start API
+        run: cd datagateway-api; nohup poetry run python -m datagateway_api.src.main > api-output.txt &
+      - name: Back a dir
+        run: cd ..
```

Comment:
This step might not be needed. I remember I added this because I was experimenting with moving between directories and wanted to make sure I was definitely in the home dir before executing any further commands. I'll try and double check this locally